kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluatorSuite.scala

```diff
@@ -253,6 +255,65 @@ class DefaultExpressionEvaluatorSuite extends AnyFunSuite with ExpressionSuiteBa
     checkBooleanVectors(actOutputVector, expOutputVector)
   }
 
+  test("evaluate expression: coalesce long") {
```

Comment:
The test code is getting repeated for each type. One rough pseudo code is here which can test all types in one-go. Feel free to ask if you have any questions. Some of the test refactoring/additions can come in separate PR before this PR.

```
Seq(ByteType.BYTE, ShortType.Short, ... (all other types) ...).foreach(type =>
  test(s"coalesce: $type") {
    // Need to update the `testColumnVector` to take an addition parameter
   // `nullAtEveryNth` which basically generates null values at every nth index.
     val col1 = testColumnVector(size = 25, dataType = type, nullAtEveryNth = 2)
     val col2 = testColumnVector(size = 25, dataType = type, nullAtEveryNth = 4)
     val col3 = testColumnVector(size = 25, dataType = type, nullAtEveryNth = 5) 

      val schema = ... (same as the code you have)
     val batch = ... (same as the code you have)

     val actResult = ... evaluator(schema, batch, coaleseExpr, type)..
     
      Seq.range(0, 25).foreach(index -> 
         // find which one is the first non-null from col1, col2, col
         // use `testColumnValue` to fetch the value as generic and compare
      )
  }
)
```

Reply:
Hi @vkorukanti , thanks for the code review , I think its a great idea to make the test more generic so that there is less copy pasta, thank you for pointing that out, I have submitted a patch that uses one single test logic but tests most of the number types via 
```
Seq(ByteType.BYTE, ShortType.Short, ... (all other types) ...).foreach(type =>
```

I am however a bit confused on the suggestions on `nullAtEveryNth` and 
```
      Seq.range(0, 25).foreach(index -> 
         // find which one is the first non-null from col1, col2, col
         // use `testColumnValue` to fetch the value as generic and compare
      )
```
Do you want to randomized the position of null, and have logic to find which col have the first non null and match that result with the output of coalesce?

Reply:
> I am however a bit confused on the suggestions on nullAtEveryNth and

Basically the current `testColumnVector` returns a test vector for a given type with null values at random indexes. By updating it to take another argument `nullAtEveryNth` we are forcing the `testColumnVector` to return the vector with nulls at fixed positions. Once you have the vectors, the next step is finding the expected result. For that, one approach is to go through the vectors to find the find non-null at the given index.

```
val nonNullValue = Seq(col1, col2, col3)
   .find(col => !col1.isNull(rowId))
   .map(col => DefaultKernelTestUtils.getValuesAsObject(col, rowId))
   .orNull
```

Reply:
Hi @vkorukanti , thanks for the reply and clarification, and apologies on the delay , was on vacation :) 
I have added the change requested but I parametrize valueAtEveryNth  instead of nullAtEveryNth
Because nullAtEveryNth gives 

    col1 = [1,n,1,n,1,n,1,n,1,n]
    col2 = [2,2,n,2,2,n,2,2,n,2]
    col3 = [5,5,5,5,n,5,5,5,5,5]

and valueAtEveryNth gives

    col1 = [n,2,n,2,n,2,n,2,n,2]
    col2 = [n,n,3,n,n,3,n,n,3,n]
    col3 = [n,n,n,n,5,n,n,n,n,5]

the latter allow us to test if null exist in every row and needs to output null as the coalesce value for that column
thanks